fluent: docs: update grep for style (#1458)
fluent/fluent-bit-docs: pipeline/filters/grep.md

---
description: Select or exclude records using patterns
---

# Grep

The _Grep Filter_ plugin lets you match or exclude specific records based on
regular expression patterns for values or nested values.

## Configuration parameters

The plugin supports the following configuration parameters:

| Key          | Value Format | Description |
| ------------ | ------------ | ----------- |
| `Regex`      | KEY REGEX | Keep records where the content of KEY matches the regular expression. |
| `Exclude`    | KEY REGEX | Exclude records where the content of KEY matches the regular expression. |
| `Logical_Op` | Operation | Specify a logical operator:  `AND`, `OR` or `legacy` (default). In `legacy` mode the behaviour is either `AND` or `OR` depending on whether the `grep` is including (uses AND) or excluding (uses OR). Available from 2.1 or higher. |

### Record Accessor Enabled

Enable the [Record Accessor](../../administration/configuring-fluent-bit/classic-mode/record-accessor.md)

> feature to specify the KEY. Use the record accesso to match values against nested

Reviewer: ```suggestion
feature to specify the KEY. Use the record accessor to match values against nested
```

{"log": "ggg"}
```

### Command line

> When using the command line, pay special attention to quote the regular expressions.

Reviewer: ```suggestion
When using the command line, pay close attention to quote the regular expressions.
```

Author: i thought i got that!.

Using a configuration file might be easier.

The following command loads the [tail](../../pipeline/inputs/tail) plugin and

> reads the content of `lines.txt` file. Then the `grep` filter applies a regular

Reviewer: ```suggestion
reads the content of `lines.txt`. Then the `grep` filter applies a regular
```
(or "the `lines.txt` file"?)

          match: '*'

```

{% endtab %}
{% endtabs %}

> The filter allows you to use multiple rules which are applied in order. You can

Reviewer: ```suggestion
The filter lets you use multiple rules which are applied in order. You can
```
